src/api/nodeRpcCall.js

```diff
@@ -774,37 +766,24 @@ const submitProposal = async (walletAddress, values, callback) => {
     additionalMetadata: {},
     proposerAddress: walletAddress,
   });
-  const legislationIndex = centralizedMetadata.data.id;
-  const hash = await getProposalHash(values, legislationIndex);
-  const notePreimageTx = api.tx.preimage.notePreimage(hash.extrinsicEncoded);
+  const extrinsicMethod = api.tx.liberlandLegislation.addLaw(
+    parseInt(values.legislationTier), 
+    centralizedMetadata.data.id, 
+    values.legislationContent
+  ).method;
+
   const minDeposit = api.consts.democracy.minimumDeposit;
   const proposeCall = parseInt(values.legislationTier) === 0 ? api.tx.democracy.proposeRichOrigin : api.tx.democracy.propose;
-  const proposeTx = proposeCall({ Legacy: hash.encodedHash }, minDeposit);
-  notePreimageTx.signAndSend(walletAddress, { signer: injector.signer }, ({ status }) => {
-    if (status.isInBlock) {
-      // eslint-disable-next-line no-console
-      console.log(`Completed NOTEPREIMAGE at block hash #${status.asInBlock.toString()}`);
-      proposeTx.signAndSend(walletAddress, { signer: injector.signer }, ({ status, events, dispatchError }) => {
-        let errorData = handleMyDispatchErrors(dispatchError, api)
-        if (status.isInBlock) {
-          // eslint-disable-next-line no-console
-          console.log(`Completed PROPOSE at block hash #${status.asInBlock.toString()}`);
-          callback(null, {
-            blockHash: status.asInBlock.toString(),
-            errorData
-          });
-        }
-      }).catch((error) => {
-        // eslint-disable-next-line no-console
-        console.error(':( transaction PROPOSE failed', error);
-        callback({isError: true, details: error.toString()});
-      });
-    }
-  }).catch((error) => {
-    // eslint-disable-next-line no-console
-    console.error(':( transaction failed', error);
-    callback({isError: true, details: error.toString()});
-  });
+
+  const existingPreimages = await api.query.preimage.preimageFor([extrinsicMethod.hash, extrinsicMethod.encodedLength])
```

Comment:
```suggestion
  const existingPreimage = await api.query.preimage.preimageFor([extrinsicMethod.hash, extrinsicMethod.encodedLength])
```
But TBH I would reduce it to just `preimage`